src/components/ethContracts.tsx

```diff
@@ -1,144 +1,87 @@
+  const [customDefsStr, setCustomDefsStr] = useState("");
+  const [success, setSuccess] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [tab, setTab] = useState(TAB_KEYS.PACK);
+  const { CAN_VIEW_CONTRACTS } = useFeature(session);
 
-  onTabChange(key) {
-    this.setState({ tab: key, success: false, error: null, loading: false });
+  function setDefaultState() {
+    setDefs([]);
+    setSuccess(false);
+    setLoading(false);
+    setCustomDefs([]);
+    setCustomDefsStr("");
   }
 
-  onSmartContractAddress(input) {
-    if (
-      input.slice(0, 2) !== "0x" ||
-      false === /^[0-9a-fA-F]+$/.test(input.slice(2)) ||
-      input.length !== 42
-    ) {
-      // Not a valid address
-      this.setState({
-        error: "Invalid Ethereum contract address",
-        ...defaultState,
-      });
-    } else {
-      this.setState({ loading: true });
-      setTimeout(() => {
-        fetch(`${constants.GET_ABI_URL}${input}`)
-          .then((response) => response.json())
-          .then((resp) => {
-            // Map confusing error strings to better descriptions
-            if (resp.err === "Contract source code not verified") {
-              resp.err =
-                "Contract source code not published to Etherscan or not verified. Cannot determine data.";
-            }
-            if (resp.err) {
-              this.setState({ error: resp.err.toString(), ...defaultState });
-            } else {
-              try {
-                const result = JSON.parse(resp.result);
-                const defs = this.TMP_REMOVE_ZERO_LEN_PARAMS(
-                  this.props.session.client.parseAbi("etherscan", result, true)
-                );
-                this.setState({
-                  defs,
-                  contract: input,
-                  error: null,
-                  success: false,
-                  loading: false,
-                });
-              } catch (err) {
-                this.setState({ error: err.toString(), ...defaultState });
-              }
-            }
-          })
-          .catch((err) => {
-            this.setState({ error: err.toString(), ...defaultState });
-          });
-      }, 5000); // 1 request per 5 seconds with no API key provided
-    }
+  function onTabChange(key) {
+    setTab(key);
+    setSuccess(false);
+    setError(null);
+    setLoading(false);
   }
 
-  addDefs(skipErrors = false, defsToAdd = null) {
-    this.setState({ loading: true, error: null });
-    // Longer timeout for loading these since requests may get dropped
-    this.props.session.client.timeout = 2 * constants.ASYNC_SDK_TIMEOUT;
-    const defs = this.state.customDefs
-      ? this.state.customDefs
-      : this.state.defs;
-    this.props.session.addAbiDefs(defs, (err) => {
+  function addDefs(skipErrors = false, defsToAdd = null) {
+    setError(null);
+    setLoading(true); // Longer timeout for loading these since requests may get dropped
```

Comment:
Missing line break on this comment

Reply:
[967d1d9](https://github.com/GridPlus/lattice-manager/pull/227/commits/967d1d92d5c90eefc8e2ab7978b6545f01559bde)